router/internal/cdn/cdn.go

```diff
@@ -163,6 +167,77 @@ func (cdn *CDN) PersistedOperation(ctx context.Context, clientName string, sha25
 	return unscaped, nil
 }
 
+type RouterConfigNotFoundError interface {
+	error
+	FederatedGraphId() string
+}
+
+type routerConfigNotFoundError struct {
+	federatedGraphId string
+}
+
+func (e *routerConfigNotFoundError) FederatedGraphId() string {
+	return e.federatedGraphId
+}
+
+func (e *routerConfigNotFoundError) Error() string {
+	return fmt.Sprintf("router config of the federated graph %s not found", e.federatedGraphId)
+}
+
+func (cdn *CDN) RouterConfig(ctx context.Context, version string) (*nodev1.RouterConfig, error) {
+	routerConfigPath := fmt.Sprintf("/%s/%s/routerconfigs/latest.json",
+		cdn.organizationID,
+		cdn.federatedGraphID,
+	)
+	routerConfigURL := cdn.cdnURL.ResolveReference(&url.URL{Path: routerConfigPath})
+
+	body, err := json.Marshal(map[string]interface{}{
+		"version": version,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", routerConfigURL.String(), bytes.NewBuffer(body))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Authorization", "Bearer "+cdn.authenticationToken)
+	resp, err := cdn.httpClient.Do(req)
```

Comment:
We should use retry mechanism here. We can use hashicorp retry client.

Reply:
Done